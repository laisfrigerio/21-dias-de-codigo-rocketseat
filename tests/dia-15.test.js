const { JSDOM } = require('jsdom')
const fs = require('fs')
const path = require('path')

const html = fs.readFileSync(path.resolve(__dirname, '../dia-15/index.html'), 'utf8')

let dom
let container

describe('dia 15', () => {
  beforeEach(() => {
    dom = new JSDOM(html, { runScripts: 'dangerously' })
    container = dom.window.document.body
  })

  test('should match the e-commerce challenge', () => {
    expect(container).toMatchSnapshot();
  });
})