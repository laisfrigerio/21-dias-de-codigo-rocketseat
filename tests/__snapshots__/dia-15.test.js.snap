// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dia 15 should match the e-commerce challenge 1`] = `
<body
  class="main-container no-toggle"
  id="particles-js"
>
  
  
  <header />
  
  
  <section
    class="my-courses"
  >
    
    
    <div
      class="course-card"
    >
      
      
      <img
        alt=""
        src="./meus-cursos/1.png"
      />
      
      
      <div
        class="info"
      >
        
        
        <h2>
          Introdução ao desenvolvimento web com HTML, CSS e Javascript
        </h2>
        
        
        <ul>
          
          
          <li>
            HTML Semântico
          </li>
          
          
          <li>
            Básico de SEO
          </li>
          
          
          <li>
            Estilização
          </li>
          
          
          <li>
            Lógica de Programação
          </li>
          
          
          <li>
            Github Pages
          </li>
          
        
        </ul>
        
        
        <p
          class="price"
        >
          399
        </p>
        
        
        <button
          class="pay"
          type="button"
        >
          COMPRAR
        </button>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="course-card"
    >
      
      
      <img
        alt=""
        src="./meus-cursos/2.png"
      />
      
      
      <div
        class="info"
      >
        
        
        <h2>
          Da teoria à prática: Clean Code com exemplos utilizando o Javascript
        </h2>
        
        
        <ul>
          
          
          <li>
            Boas práticas de programação
          </li>
          
          
          <li>
            Refatoração
          </li>
          
          
          <li>
            Testes
          </li>
          
          
          <li>
            Refatorando com testes
          </li>
          
          
          <li>
            Github Actions
          </li>
          
        
        </ul>
        
        
        <p
          class="price"
        >
          599
        </p>
        
        
        <button
          class="pay"
          type="button"
        >
          COMPRAR
        </button>
        
      
      </div>
      
    
    </div>
    
    
    <div
      class="course-card"
    >
      
      
      <img
        alt=""
        src="./meus-cursos/3.png"
      />
      
      
      <div
        class="info"
      >
        
        
        <h2>
          Introdução ao desenvolvimento de teste utilizando a ferramenta JEST
        </h2>
        
        
        <ul>
          
          
          <li>
            Teste de Unidade
          </li>
          
          
          <li>
            Teste de Integração
          </li>
          
          
          <li>
            Teste de Snapshot
          </li>
          
          
          <li>
            Mock
          </li>
          
          
          <li>
            Github Actions
          </li>
          
        
        </ul>
        
        
        <p
          class="price"
        >
          499
        </p>
        
        
        <button
          class="pay"
          type="button"
        >
          COMPRAR
        </button>
        
      
      </div>
      
    
    </div>
    
  
  </section>
  
  
  <footer />
  
  
  <script
    src="https://cdn.jsdelivr.net/npm/particles.js@2.0.0/particles.min.js"
  />
  
  
  <script
    src="../assets/js/theme.js"
  />
  
  
  <script
    src="../assets/js/particles.js"
  />
  
  
  <script
    src="../assets/js/include-html.js"
  />
  
  
  <script
    src="./scripts.js"
  />
  



</body>
`;
